homeassistant/components/frigidaire/climate.py

```diff
@@ -0,0 +1,253 @@
+"""ClimateEntity for frigidaire integration."""
+from __future__ import annotations
+
+import logging
+
+import frigidaire
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_COOL,
+    HVAC_MODE_FAN_ONLY,
+    HVAC_MODE_OFF,
+    SUPPORT_FAN_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS, TEMP_FAHRENHEIT
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up frigidaire from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    def get_entities(username: str, password: str) -> frigidaire.Frigidaire:
+        client = frigidaire.Frigidaire(username, password)
+        appliances = client.get_appliances()
+        return [FrigidaireClimate(client, appliance) for appliance in appliances]
+
+    entities = await hass.async_add_executor_job(
+        get_entities, entry.data["username"], entry.data["password"]
+    )
+
+    async_add_entities(entities)
+
+
+class FrigidaireClimate(ClimateEntity):
+    """Representation of a Frigidaire appliance."""
+
+    def __init__(self, client, appliance):
+        """Build FrigidaireClimate.
+
+        data: initially-fetched data.
+        """
+
+        self._client = client
+        self._appliance = appliance
+        self._details = None
+        self._available = False
+        self.update()
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+
+    @property
+    def temperature_unit(self):
```

Comment:
This needs only to be calculated once, right? So maybe pre-calc it in the init so it doesn't get figured out each time the property is requested.

Additionally, the entity class attributes might work for this in that case as well.

Reply:
This can be changed during normal operation, so it wouldn't be correct to pre-calc it